### PR TITLE
Add 'Schedule Basic Public Post'

### PR DESCRIPTION
**Pull request by alisterscott**
@bsessions85 / @hoverduck would you mind 👀 at this one?

I'm a bit stuck as the auto-save doesn't seem to work after setting the post date to the future - which causes all sorts of problems.

I've been banging my head against my monitor trying to work this out and I'd appreciate some extra 👀 

Also, @hoverduck: do you see any reason this shouldn't work for Jetpack sites?

### CODE REVIEW

**Comment by alisterscott** [comment]
🤔 it seems to have passed CI but fails locally - even more reason to investigate

**Comment by bsessions85** [comment]
I looked at this today and it fails for me locally in Chrome 62.0.3202.75, but passes in the wp-e2e-tests image that is running Chrome 61.0.3163.79. When I changed the wp-e2e-tests image to use the same Chrome version, it failed in there also. I tried to  run it in standalone-chrome-debug and it passes on images running Chrome 62.0.3202.62 and 61.0.3163.100. I tried running locally and at https://wordpress.com, and the results were consistent. So all that to say I am not sure what is going on or where the trouble lies, but it seems to be related to the latest Chrome updates.

**Comment by alisterscott** [comment]
I can reproduce the editor state on another test site in all browsers - and even using our e2e account in different browsers so I'm not sure it's a Chrome thing but an account thing.

That doesn't explain why this passes in CircleCI though unless we're using a different account.

File: specs/wp-post-editor-spec.js [hunk 1]
```diff
@@ -343,6 +344,69 @@ test.describe( `[${host}] Editor: Posts (${screenSize})`, function() {
...
+					publishDate = publishDateShown;
+				} );
+				postEditorSidebarComponent.closeStatusSection();
+				postEditorToolbarComponent = new PostEditorToolbarComponent( driver );
+				postEditorToolbarComponent.ensureSaved( { clickSave: true } );
+				return postEditorToolbarComponent.clickPublishPost();
+			} );
+
+			test.it( 'Can confirm scheduling post and see correct publish date', function() {
+				let editorConfirmationSidebarComponent = new EditorConfirmationSidebarComponent( this.driver );
```
Comment (rachelmcr) [inline specs/wp-post-editor-spec.js]: Should `this.driver` in this test block be `driver`? When I make that change this scheduled post test passes locally, without any trouble caused by a dirty editor state or other issues.

Comment (alisterscott) [reply specs/wp-post-editor-spec.js]: Thanks, it makes sense for those not to contain `this`, but changing those the test still fails locally for me :(

**Comment by brbrr** [comment]
I gave it a try and was able to reproduce it manually on both `dev` & `staging` envs. After saving scheduled post - 'Saved' text did not appear. But did not managed to understand what the problem is :/

**Comment by alisterscott** [comment]
https://2392-102822243-gh.circle-artifacts.com/0/home/ubuntu/wp-e2e-tests-for-branches/wp-e2e-tests/screenshots/1516687198171-FAILED-EN-MOBILE-can-log-in.png

**Comment by alisterscott** [comment]
Now failing on CircleCI which is consistent with local runs

**Comment by alisterscott** [comment]
This wp-calypso PR: https://github.com/Automattic/wp-calypso/pull/21944 fixes this problem 🎉 

So it is a genuine problem with Calypso after all :)

**Comment by alisterscott** [comment]
This is failing due to the bug which only effects sites set to UTC+0

**Comment by alisterscott** [comment]
@brbrr would you mind reviewing this (again) now that it's working :)

**Comment by brbrr** [comment]
It is working great on desktop, but unfortunately failing when I run it w/ mobile `BROWSERSIZE`:

```
$ BROWSERSIZE=mobile ./node_modules/.bin/mocha ./specs/wp-post-editor-spec.js -g "Schedule Basic Public Post"

[WPCOM] Editor: Posts (mobile) Schedule Basic Public Post @parallel @jetpack Schedule a New Post Can schedule content for a future date (first day of second week next month):
     TimeoutError: Timed out waiting for element with css selector of 'span.editor-ground-control__save-status[data-e2e-status="Saved"]' to be present and displayed
```

**Comment by alisterscott** [comment]
@brbrr: 🤔 I thought I had fixed that with my latest commit:

```
Alisters-iMac:wp-e2e-tests alisterscott$ BROWSERSIZE=mobile ./node_modules/.bin/mocha ./specs/wp-post-editor-spec.js -g "Schedule Basic Public Post"


  [WPCOM] Editor: Posts (mobile)
    Schedule Basic Public Post @parallel @jetpack
      ✓ Delete Cookies and Local Storage
      Schedule a New Post
        ✓ Can log in (21249ms)
        ✓ Can enter post title and content (3825ms)
        ✓ Can schedule content for a future date (first day of second week next month) (11177ms)
        ✓ Can confirm scheduling post and see correct publish date (1892ms)


  5 passing (42s)
```
Did you pull the latest changes?

**Comment by brbrr** [comment]
My bad - forgot to pull the changes

:shipit: 👍

**Pull request merged**